libgnucash/engine/gnc-budget.cpp

```diff
@@ -523,18 +560,22 @@ gnc_budget_set_account_period_value(GncBudget *budget, const Account *account,
     g_return_if_fail (budget != NULL);
     g_return_if_fail (account != NULL);
 
-    make_period_path (account, period_num, path_part_one, path_part_two);
+    auto& perioddata = get_perioddata (budget, account, period_num);
+    auto budget_kvp { QOF_INSTANCE (budget)->kvp_data };
+    auto path = make_period_data_path (account, period_num);
 
     gnc_budget_begin_edit(budget);
     if (gnc_numeric_check(val))
-        qof_instance_set_kvp (QOF_INSTANCE (budget), NULL, 2, path_part_one, path_part_two);
+    {
+        delete budget_kvp->set_path (path, nullptr);
+        perioddata.value_is_set = false;
+    }
     else
     {
-        GValue v = G_VALUE_INIT;
-        g_value_init (&v, GNC_TYPE_NUMERIC);
-        g_value_set_boxed (&v, &val);
-        qof_instance_set_kvp (QOF_INSTANCE (budget), &v, 2, path_part_one, path_part_two);
-        g_value_unset (&v);
+        KvpValue* v = new KvpValue (*(gnc_numeric*)(&val));
```

Comment:
What the heck is `(*(gnc_numeric*)(&val))` supposed to do?

Reply:
It's copied directly from https://github.com/Gnucash/gnucash/blob/831b25f8f8a218e5e9b75bcc6c0161e8ba1bd6b1/libgnucash/engine/kvp-frame.cpp#L347

Reply:
this is now simplified.

Reply:
> It's copied directly from 

OK, but there's a big difference between the `void*` return value of `g_value_get_boxed` and the `gnc_numeric` passed in by value in `val`.